lib/elixir/src/elixir_errors.erl

```diff
@@ -50,7 +62,7 @@ print_diagnostic(#{severity := Severity, message := Message, stacktrace := Stack
   io:put_chars(standard_error, [prefix(Severity), Message, Location, "\n\n"]),
   Diagnostic.
 
-emit_diagnostic(Severity, Position, File, Message, Stacktrace) ->
+emit_diagnostic(Severity, Position, File, Message, Stacktrace, IsFancy) ->
```

Comment:
This option is not about being fancy or not but rather, it is an option if we should read the snippet or not.

Reply:
If we cannot read the snippet, then we still show it fancy, but without the snippet.

Reply:
That makes sense, thanks for the feedback! I've updated the code so now `Code.print_diagnostic` reads the snippet by default and `:elixir_errors.print_diagnostic` always uses `format_snippet` but the caller can decide whether to read the snippet

Reply:
We can have an option called snippet: true | false.

Reply:
Don't forget to rename this variable. :)

Reply:
```suggestion
emit_diagnostic(Severity, Position, File, Message, Stacktrace, ReadSnippet) ->
```